src/kernels/raw/launcher/kernelLauncher.node.ts

```diff
@@ -312,3 +225,63 @@ export class KernelLauncher implements IKernelLauncher {
         };
     }
 }
+
+/**
+ * Sometimes users install this in user site_packages and things don't work as expected.
+ * It should be installed into the specific python env.
+ * Logging this information would be helpful in diagnosing issues.
+ */
+async function logIPyKernelPath(
+    resource: Resource,
+    kernelConnectionMetadata: LocalKernelSpecConnectionMetadata | PythonKernelConnectionMetadata,
+    pythonExecFactory: IPythonExecutionFactory,
+    token: CancellationToken
+) {
+    const interpreter = kernelConnectionMetadata.interpreter;
+    if (
+        !isLocalConnection(kernelConnectionMetadata) ||
+        !isPythonKernelConnection(kernelConnectionMetadata) ||
+        !interpreter
+    ) {
+        return;
+    }
+    const service = await pythonExecFactory.createActivatedEnvironment({
+        interpreter,
+        resource
+    });
+    const output = await service.exec(
+        [
+            '-c',
+            'import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)'
```

Comment:
You could put this guid in a constant. Also if you don't generate a new guid is there a risk of someone injecting their own to break the service?